### ISSUE TITLE
doesn't build gem from clean

### ISSUE THREAD

**Issue by zenspider** [BrendanThompson/rb-scpt#19]
Each version is doubling in size https://rubygems.org/gems/rb-scpt

Because each version has the previous version gem installed inside it.

Either it needs to build from a fresh gem clone, or the gemspec should have a manifest of what only/actually belongs in it.

**Comment by zenspider** [comment]
This should be addressed by #20

**Comment by zenspider** [comment]
```
609 $ rake clean default
rm -f 
mkdir -p tmp/arm64-darwin23/AE/3.3.0
cd tmp/arm64-darwin23/AE/3.3.0
/Users/ryan/.rubies/ruby-3.3.0/bin/ruby -I. ../../../../ext/AE/extconf.rb
creating Makefile
cd -
cd tmp/arm64-darwin23/AE/3.3.0
/usr/bin/make
compiling ../../../../ext/AE/AE.c
../../../../ext/AE/AE.c:62:1: warning: function 'rbAE_raiseMacOSError' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
   62 | {
      | ^
../../../../ext/AE/AE.c:186:35: warning: implicit conversion loses integer precision: 'long' to 'AETransactionID' (aka 'int') [-Wshorten-64-to-32]
  186 |   AETransactionID transactionID = NUM2LONG(transactionIDValue);
      |                   ~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ryan/.rubies/ruby-3.3.0/include/ruby-3.3.0/ruby/internal/arithmetic/long.h:51:22: note: expanded from macro 'NUM2LONG'
   51 | #define NUM2LONG     RB_NUM2LONG          /**< @old{RB_NUM2LONG} */
      |                      ^
/Users/ryan/.rubies/ruby-3.3.0/include/ruby-3.3.0/ruby/internal/arithmetic/long.h:57:22: note: expanded from macro 'RB_NUM2LONG'
   57 | #define RB_NUM2LONG  rb_num2long_inline   /**< @alias{rb_num2long_inline} */
      |                      ^
../../../../ext/AE/AE.c:249:41: warning: implicit conversion loses integer precision: 'Size' (aka 'long') to 'int' [-Wshorten-64-to-32]
  249 |                                 INT2NUM(dataSize)
      |                                 ~~~~~~~ ^~~~~~~~
../../../../ext/AE/AE.c:329:18: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
  329 |   return INT2NUM(length);
      |          ~~~~~~~ ^~~~~~
../../../../ext/AE/AE.c:494:9: warning: 'LSFindApplicationForInfo' is deprecated: first deprecated in macOS 10.10 - Use LSCopyApplicationURLsForBundleIdentifier instead. [-Wdeprecated-declarations]
  494 |   err = LSFindApplicationForInfo(inCreator,
      |         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LSInfoDeprecated.h:765:1: note: 'LSFindApplicationForInfo' has been explicitly marked deprecated here
  765 | LSFindApplicationForInfo(
      | ^
../../../../ext/AE/AE.c:502:9: warning: 'FSRefMakePath' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  502 |   err = FSRefMakePath(&outAppRef, path, PATH_MAX);
      |         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4086:18: note: 'FSRefMakePath' has been explicitly marked deprecated here
 4086 | extern OSStatus  FSRefMakePath(const FSRef *ref, UInt8 *path, UInt32 pathBufferSize)     __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
      |                  ^
../../../../ext/AE/AE.c:515:9: warning: 'FSPathMakeRef' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  515 |   err = FSPathMakeRef((UInt8 *)StringValuePtr(path), &appRef, NULL);
      |         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4115:18: note: 'FSPathMakeRef' has been explicitly marked deprecated here
 4115 | extern OSStatus  FSPathMakeRef(const UInt8 *path, FSRef *ref, Boolean *isDirectory)        __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
      |                  ^
../../../../ext/AE/AE.c:518:11: warning: 'GetNextProcess' is deprecated: first deprecated in macOS 10.9 [-Wdeprecated-declarations]
  518 |     err = GetNextProcess(&psn);
      |           ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/ApplicationServices.framework/Frameworks/HIServices.framework/Headers/Processes.h:375:1: note: 'GetNextProcess' has been explicitly marked deprecated here
  375 | GetNextProcess(ProcessSerialNumber * pPSN)                     AVAILABLE_MAC_OS_X_VERSION_10_0_AND_LATER_BUT_DEPRECATED_IN_MAC_OS_X_VERSION_10_9;
      | ^
../../../../ext/AE/AE.c:520:11: warning: 'GetProcessBundleLocation' is deprecated: first deprecated in macOS 10.9 [-Wdeprecated-declarations]
  520 |     err = GetProcessBundleLocation(&psn, &foundRef);
      |           ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/ApplicationServices.framework/Frameworks/HIServices.framework/Headers/Processes.h:662:1: note: 'GetProcessBundleLocation' has been explicitly marked deprecated here
  662 | GetProcessBundleLocation(
      | ^
../../../../ext/AE/AE.c:522:9: warning: 'FSCompareFSRefs' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  522 |     if (FSCompareFSRefs(&appRef, &foundRef) == noErr)
      |         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:1850:15: note: 'FSCompareFSRefs' has been explicitly marked deprecated here
 1850 | extern OSErr  FSCompareFSRefs(const FSRef *ref1, const FSRef *ref2)                     __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
      |               ^
../../../../ext/AE/AE.c:534:9: warning: 'GetProcessForPID' is deprecated: first deprecated in macOS 10.9 [-Wdeprecated-declarations]
  534 |   err = GetProcessForPID(NUM2INT(pid), &psn);
      |         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/ApplicationServices.framework/Frameworks/HIServices.framework/Headers/Processes.h:762:1: note: 'GetProcessForPID' has been explicitly marked deprecated here
  762 | GetProcessForPID(
      | ^
../../../../ext/AE/AE.c:547:9: warning: 'FSPathMakeRef' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  547 |   err = FSPathMakeRef((UInt8 *)StringValuePtr(path), &appRef, NULL);
      |         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4115:18: note: 'FSPathMakeRef' has been explicitly marked deprecated here
 4115 | extern OSStatus  FSPathMakeRef(const UInt8 *path, FSRef *ref, Boolean *isDirectory)        __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
      |                  ^
../../../../ext/AE/AE.c:549:3: warning: 'LSApplicationParameters' is deprecated: first deprecated in macOS 10.10 - Use NSWorkspace instead. [-Wdeprecated-declarations]
  549 |   LSApplicationParameters appParams = {0,
      |   ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LSOpenDeprecated.h:153:3: note: 'LSApplicationParameters' has been explicitly marked deprecated here
  153 | } LSApplicationParameters API_DEPRECATED("Use NSWorkspace instead.", macos(10.4,10.10) ) API_UNAVAILABLE( ios, tvos, watchos );
      |   ^
../../../../ext/AE/AE.c:554:9: warning: 'LSOpenApplication' is deprecated: first deprecated in macOS 10.10 - Use -[NSWorkspace launchApplicationAtURL:options:configuration:error:] instead. [-Wdeprecated-declarations]
  554 |   err = LSOpenApplication(&appParams, &psn);
      |         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LSOpenDeprecated.h:185:1: note: 'LSOpenApplication' has been explicitly marked deprecated here
  185 | LSOpenApplication(
      | ^
../../../../ext/AE/AE.c:710:9: warning: 'FSPathMakeRef' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  710 |   err = FSPathMakeRef((UInt8 *)StringValuePtr(path), &fsRef, NULL);
      |         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4115:18: note: 'FSPathMakeRef' has been explicitly marked deprecated here
 4115 | extern OSStatus  FSPathMakeRef(const UInt8 *path, FSRef *ref, Boolean *isDirectory)        __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
      |                  ^
15 warnings generated.
compiling ../../../../ext/AE/SendThreadSafe.c
linking shared-object AE.bundle
cd -
mkdir -p tmp/arm64-darwin23/stage/lib/AE
/usr/bin/make install sitearchdir=../../../../lib/AE sitelibdir=../../../../lib/AE target_prefix=
/usr/bin/install -c -m 0755 AE.bundle ../../../../lib/AE
cp tmp/arm64-darwin23/AE/3.3.0/AE.bundle tmp/arm64-darwin23/stage/lib/AE/AE.bundle
/Users/ryan/Work/git/BrendanThompson/rb-scpt/lib/_aem/aemreference.rb:34: warning: undefining the allocator of T_DATA class AE::AEDesc
Run options: --seed 64595

# Running:

...............S...............

Finished in 0.924778s, 33.5216 runs/s, 432.5362 assertions/s.

31 runs, 400 assertions, 0 failures, 0 errors, 1 skips

You have skipped tests. Run with --verbose for details.
610 $ 
```
